src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivoreObj) {
+    if (herbivoreObj instanceof Herbivore && herbivoreObj.hidden === false) {
+      herbivoreObj.health -= 50;
+      Animal.alive = Animal.alive.filter((el) => el.health > 0);
```

Comment:
Instead of filtering the whole array every time a bite occurs, you could check if the herbivore has died, and if so, filter the array.
That way you would remove many unnecessary filters (at least one filter for every herbivore and all filters when a carnivore attacks another)